cumulusci/tasks/github/merge.py

```diff
@@ -15,57 +15,187 @@ class MergeBranch(BaseGithubTask):
-        source_branch = self.options.get('source_branch', self.project_config.project__git__default_branch)
+        if 'commit' not in self.options:
+            self.options['commit'] = self.project_config.repo_commit
+        if 'branch_prefix' not in self.options:
+            self.options['branch_prefix'] = self.project_config.project__git__prefix_feature
+        if 'source_branch' not in self.options:
+            self.options['source_branch'] = self.project_config.project__git__default_branch
+        if 'children_only' not in self.options:
+            self.options['children_only'] = False
+        elif self.options['children_only'] in [True, 'True']:
+            self.options['children_only'] = True
+
+    def _run_task(self):
+        self.repo = self.get_repo()
 
-        head_branch = repo.branch(source_branch)
+        head_branch = self.repo.branch(self.options['source_branch'])
         if not head_branch:
-            message = 'Branch {} not found'.format(source_branch)
+            message = 'Branch {} not found'.format(self.options['source_branch'])
             self.logger.error(message)
             raise GithubApiNotFoundError(message)
 
         # Get existing pull requests targeting a target branch
-        existing_prs = []
-        for pr in repo.iter_pulls(state='open'):
-            if pr.base.ref.startswith(branch_prefix):
-                existing_prs.append(pr.base.ref)
-        
-        targets = []
-        for branch in repo.iter_branches():
-            if not branch.name.startswith(branch_prefix):
-                self.logger.info('Skipping branch {}: does not match prefix {}'.format(branch.name, branch_prefix))
+        self.existing_prs = []
+        for pr in self.repo.iter_pulls(state='open'):
+            if pr.base.ref.startswith(self.options['branch_prefix']):
+                self.existing_prs.append(pr.base.ref)
+       
+        # Create list and dict of all target branches 
+        branches = []
+        branches_dict = {}
+        for branch in self.repo.iter_branches():
+            if branch.name == self.options['source_branch']:
+                if not self.options['children_only']:
+                    self.logger.debug('Skipping branch {}: is source branch'.format(branch.name))
+                    branches_dict[branch.name] = branch
+                    continue
+            if not branch.name.startswith(self.options['branch_prefix']):
+                if not self.options['children_only']:
+                    self.logger.debug('Skipping branch {}: does not match prefix {}'.format(branch.name, self.options['branch_prefix']))
                 continue
-            if branch.name == source_branch:
-                self.logger.info('Skipping branch {}: is source branch'.format(branch.name))
+            branches.append(branch)
+            branches_dict[branch.name] = branch
+
+        # Identify parent/child branches
+        possible_children = []
```

Comment:
Might be nice to encapsulate the building of possible parents/children into another function. Or go big and encapsulate all of building the `branch_tree`.

Reply:
Went even bigger and encapsulated almost all of _run_task :)